provider/pkg/gen/schema.go

```diff
@@ -433,9 +433,20 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 		}
 	}
 
+	description := dd.createMethod.Description
+
+	// Apply auto-naming.
+	namePattern, err := namePropertyPattern(inputProperties)
+	if err == nil {
```

Comment:
I think it might be clearer to include the HasSuffix condition as an OR on the first branch, and then use a normal `else` branch. I'm also not quite clear on why IamPolicy is special, so you might consider including that condition in a named function.

Reply:
Refactored to a named variable and an explicit if statement on top